cmd/config-migrate.go

```diff
@@ -2413,8 +2413,58 @@ func migrateV27ToV28() error {
 	return nil
 }
 
+// Migrates ${HOME}/.minio/config.json to '<export_path>/.minio.sys/config/config.json'
+func migrateConfigToMinioSys(objAPI ObjectLayer) error {
+	defer os.Rename(getConfigFile(), getConfigFile()+".deprecated")
```

Comment:
Should be renamed only if:
1) checkConfig returns nil
OR
2) saveServerConfig() returns nil